cli/container.go

```diff
@@ -90,6 +91,71 @@ func (c *container) config() (*types.ContainerCreateConfig, error) {
 			config.HostConfig.MemorySwap = v
 		}
 	}
+	// parse device mappings
+	deviceMappings := []*types.DeviceMapping{}
+	for _, device := range c.devices {
+		deviceMapping, err := parseDevice(device)
+		if err != nil {
+			return nil, fmt.Errorf("parse devices error: %s", err)
+		}
+		deviceMappings = append(deviceMappings, deviceMapping)
+	}
+	config.HostConfig.Devices = deviceMappings
 
 	return config, nil
 }
+
+// parseDevice parses a device mapping string to a container.DeviceMapping struct
+func parseDevice(device string) (*types.DeviceMapping, error) {
```

Comment:
I think it is better to add some unit tests for function `parseDevice` and `validDeviceMode`.

In addition, this validation is good in client side, while maybe in the future we should support this in daemon side. We need to validate the API.

Reply:
Yes, your suggestion is great, i will put this two function into pkg/utils, and add some unit tests